snuba/datasets/entities/discover.py

```diff
@@ -460,6 +463,7 @@ def get_query_processors(self) -> Sequence[QueryProcessor]:
             TagsExpanderProcessor(),
             BasicFunctionsProcessor(),
             ProjectRateLimiterProcessor(project_column="project_id"),
+            EmptyTagConditionProcessor(),
```

Comment:
Any reason for adding this processor to the logical layer? the existence of a bloomfilter index on the tags.key column is specific to the storage (not guaranteed to exist on a specific storage), so it seems an optimization to run on the clickhouse layer.

Though we are actually asking clients to apply that optimization on their own as well, so I guess ok?
If you decide to move to the clickhouse layer there is already a matcher for resolved tags:
https://github.com/getsentry/snuba/blob/master/snuba/clickhouse/translators/snuba/mappers.py#L235-L256

Reply:
At the moment we're not asking client to add that optimization. This is for internal testing to see if it even helps. I'll move this to the storage layer, that does make more sense.